packages/web3-eth/src/utils/decode_signed_transaction.ts

```diff
@@ -0,0 +1,47 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+import { TransactionFactory } from '@ethereumjs/tx';
+import { HexStringBytes, SignedTransactionInfoAPI, TransactionSignedAPI } from 'web3-types';
+import { bytesToHex, DataFormat, format, hexToBytes, keccak256 } from 'web3-utils';
+import { detectRawTransactionType } from './detect_transaction_type';
+import { formatTransaction } from './format_transaction';
+
+/**
+ * Decodes an [RLP](https://ethereum.org/en/developers/docs/data-structures-and-encoding/rlp/#top) encoded transaction.
+ *
+ * @param encodedSignedTransaction The RLP encoded transaction.
+ * @param returnFormat ({@link DataFormat} Specifies how the return data should be formatted.
+ * @returns {@link SignedTransactionInfoAPI}, an object containing the RLP encoded signed transaction (accessed via the `raw` property) and the signed transaction object (accessed via the `tx` property).
+ */
+export function decodeSignedTransaction<ReturnFormat extends DataFormat>(
```

Comment:
does this support EIP-2718 Enveloped transactions decoding? I think not,  Its only for RLP encoded 0x0 tx decoding?

Reply:
Yes it can handle type transactions as shown by tests still passing after [this](https://github.com/ChainSafe/web3.js/pull/5312/commits/ec538fdc8b4e3844c391ec3ebe18d388a9bd497f) commit. This is because transaction parsing is done via `@ethereumjs/tx`'s `TransactionFactory.fromSerializedData` [here](https://github.com/ChainSafe/web3.js/blob/ec538fdc8b4e3844c391ec3ebe18d388a9bd497f/packages/web3-eth/src/utils/decode_signed_transaction.ts#L38)